### ISSUE TITLE
Add Shell Upgrade for Linux Payloads

### ISSUE THREAD

**Issue by Thy-GoD** [t3l3machus/Villain#112]
I've discovered that the Linux Rev Shells aren't capable of being upgraded to a full shell, so things like `sudo -l` and tab completion don't work.

`Ctrl + Z` and trying the shell upgrade like usual doesn't work either.

Unless i'm missing something, are there plans to include this feature?

**Comment by t3l3machus** [comment]
Not like in the top priority list, but yeah, definitely!

**Comment by Thy-GoD** [comment]
> Not like in the top priority list, but yeah, definitely!

Thanks for the reply, and no worries in priority, I really like this tool's idea!

**Comment by DocMors** [comment]
In the python shell this works very well. check https://pastebin.com/kb3a4SGx

**Comment by Thy-GoD** [comment]
> In the python shell this works very well. check https://pastebin.com/kb3a4SGx

seems like a priv escalation and not an actual dumb shell upgrade.

**Comment by Mag1cByt3s** [comment]
I am having the same issue.

Trying to run interactive tools on linux machines fails because the shell is not a full PTY.

**Comment by Thy-GoD** [comment]
> I am having the same issue.
> 
> Trying to run interactive tools on linux machines fails because the shell is not a full PTY.

Yeah unfortunately you'd have to get another shell, i dont actually use villain anymore due to this exact issue, I only ever use it for windows machines to generate payloads for me.

**Comment by Mag1cByt3s** [comment]
> > I am having the same issue.
> > Trying to run interactive tools on linux machines fails because the shell is not a full PTY.
> 
> Yeah unfortunately you'd have to get another shell, i dont actually use villain anymore due to this exact issue, I only ever use it for windows machines to generate payloads for me.

May I ask what C2 you ended up using now?

**Comment by Thy-GoD** [comment]
> > > I am having the same issue.
> > > Trying to run interactive tools on linux machines fails because the shell is not a full PTY.
> > 
> > 
> > Yeah unfortunately you'd have to get another shell, i dont actually use villain anymore due to this exact issue, I only ever use it for windows machines to generate payloads for me.
> 
> May I ask what C2 you ended up using now?

I don't use any C2, if i were to use one i'd probs use msfconsole.

I havn't gotten to try C2s like cobalt strike or empire yet, but msfconsole is probs the closest one i'd use.

If you ever find alternative to villain for this purpose, it'd help alot.

Also i believe it should be possible to recycle my tmux shell idea in [here](https://github.com/t3l3machus/Villain/pull/116) but that'd just be extra steps.
